lib/github_client.rb

```diff
@@ -0,0 +1,30 @@
+require_relative './json_http_client.rb'
+
+# +GitHubClient+ interfaces with the GitHub REST API.
+#
+# @see https://docs.github.com/en/free-pro-team@latest/rest
+class GitHubClient
+
+  # A new instance of +GitHubClient+
+  #
+  # @param base_url [String] the base URL to use when calling the GitHub API
+  def initialize(base_url: 'https://api.github.com')
+    @base_url = base_url
+    @json_http_client = JSONHTTPClient.new
+  end
+
+  # Checks whether the given pull request in the given repository is closed
+  #
+  # @param repo [String] the name of the GitHub repository with the "org/name" format, e.g. "Automattic/bintray-garbage-collector"
+  # @param pr_id [Integer] the pull request id
+  # @return [Bool] whether the pull request is closed or nil if the request fails
```

Comment:
ditto, wondering if we shouldn't just return `false` if request fails to avoid optional bool?